### PR TITLE
fix: consider using createroot instead of render

### PR DESCRIPTION
**Pull request by harish-sethuraman**
**What kind of change does this PR introduce?**

I see that there is a react template support being added so used `createRoot` instead of `render` as per React 18.

**Did you add tests for your changes?**

### CODE REVIEW

**Review by snitin315 — CHANGES_REQUESTED** [review]
The scaffolded app can be using react 17?

**Comment by harish-sethuraman** [comment]
I don't see any version specification. Just the package name and this `doInstall` util installs all packages. So it pulls in the latest version?

- https://github.com/webpack/webpack-cli/blob/888bed93f2cbe348c63afc32d6dab6ecd02b7c47/packages/webpack-cli/src/webpack-cli.ts#L272-L273

**Comment by snitin315** [comment]
Yeah, it will use the latest version. Ideally, we should specify the version because it is possible that the current template is not compatible with the latest version.

**Comment by harish-sethuraman** [comment]
https://github.com/webpack/webpack-cli/blob/master/packages/generators/init-template/react/package.json.tpl

Although we add react version to the scaffold when we `doInstall` using the util the latest version only will be fetched right?

**Comment by alexander-akait** [comment]
We have react here https://github.com/webpack/webpack-cli/blob/master/packages/generators/src/handlers/react.ts#L28, just need to look how to provide version

**Comment by harish-sethuraman** [comment]
Done I see that this works when we push deps with the version as @snitin315 mentioned

File: packages/generators/src/handlers/react.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export async function questions(
   });
 
   // Add react dependencies
-  self.dependencies.push("react", "react-dom");
+  self.dependencies.push("react@18.1.0", "react-dom@18.1.0");
```
Comment (evenstensberg) [inline packages/generators/src/handlers/react.ts:28]: I don't think we should pin specific version numbers

Comment (alexander-akait) [reply packages/generators/src/handlers/react.ts:28]: Yep, just `react@18`

Comment (harish-sethuraman) [reply packages/generators/src/handlers/react.ts]: Done 👍🏻

**Review by rishabh3112 — APPROVED** [review]
Looks fine now, Lets see CI then we can merge.

**Review by alexander-akait — APPROVED** [review]

**Comment by harish-sethuraman** [comment]
Bumping for merge :)

**Review by snitin315 — APPROVED** [review]
Thanks!

**Pull request merged**